include/bm/bm_sim/fanout_pkt_mgr.h

```diff
@@ -0,0 +1,112 @@
+/* Copyright 2025-present Contributors to the P4 Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef BM_BM_SIM_FANOUT_PKT_MGR_H_
+#define BM_BM_SIM_FANOUT_PKT_MGR_H_
+
+#include <memory>
+#include <unordered_map>
+#include <vector>
+#include "logger.h"
+#include "packet.h"
+#include "match_tables.h"
+#include "action_profile.h"
+
+namespace bm {
+class MatchTableIndirect;
+using bm::ActionProfile;
+using EntryVec = const std::vector<const ActionEntry*>;
+using SelectorIface = ActionProfile::GroupSelectionIface;
+
+struct FanoutCtx {
+  bool hit{false};
+  const Packet * cur_pkt{nullptr};
+  ActionProfile *action_profile{nullptr};
+  MatchTableIndirect *cur_table{nullptr};
+  std::function<void(const bm::Packet *)> buffer_push_fn;
+
+  explicit FanoutCtx(
+    const std::function<void(const bm::Packet *)> &buffer_push_fn)
+      : buffer_push_fn(buffer_push_fn) { }
+};
+
+class FanoutPktSelection: public SelectorIface{
+ public:
+    using grp_hdl_t = ActionProfile::grp_hdl_t;
+    using mbr_hdl_t = ActionProfile::mbr_hdl_t;
+    using hash_t = ActionProfile::hash_t;
+    using MatchErrorCode = bm::MatchErrorCode;
+
+    FanoutPktSelection() = default;
+
+    // callbacks after member op, not actual member/group ops
+    void add_member_to_group(grp_hdl_t grp, mbr_hdl_t mbr) override;
+
+    void remove_member_from_group(grp_hdl_t grp, mbr_hdl_t mbr) override;
+
+    mbr_hdl_t get_from_hash(grp_hdl_t grp, hash_t h) const override;
+
+    void reset() override {}
+
+ private:
+    std::unordered_map<grp_hdl_t, std::vector<mbr_hdl_t>> groups;
+};
+
+class FanoutPktMgr {
```

Comment:
Let's add some documentation regarding the FanoutPktMgr. For example how it is used and use cases